GitUI/CommandsDialogs/WorktreeDialog/FormManageWorktree.cs

```diff
@@ -192,75 +160,91 @@ private enum HeadType
                 }
+            }
+        }
 
+        private void buttonOpenSelectedWorktree_Click(object sender, EventArgs e)
+        {
+            if (!CanActOnSelectedWorkspace(out var workTree))
+            {
                 return;
             }
 
-            if (e.ColumnIndex == 6)
+            if (AppSettings.DontConfirmSwitchWorktree || MessageBox.Show(this,
+                    _switchWorktreeText.Text, _switchWorktreeTitle.Text, MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
             {
-                if (e.RowIndex == 0)
+                if (Directory.Exists(workTree.Path))
                 {
-                    return;
+                    ((FormBrowse)Owner).SetWorkingDir(System.IO.Path.GetFullPath(workTree.Path));
+                    Close();
                 }
+            }
+        }
 
-                if (MessageBox.Show(this, _deleteWorktreeText.Text, _deleteWorktreeTitle.Text,
-                                    MessageBoxButtons.YesNo, MessageBoxIcon.Warning) == DialogResult.Yes)
-                {
-                    Validates.NotNull(workTree.Path);
+        private void Worktrees_SelectionChanged(object sender, EventArgs e)
+        {
+            buttonDeleteSelectedWorktree.Enabled = CanDeleteSelectedWorkspace();
+            buttonOpenSelectedWorktree.Enabled = CanActOnSelectedWorkspace(out _);
+        }
 
-                    if (workTree.Path.TryDeleteDirectory(out string? errorMessage))
-                    {
-                        PruneWorktrees();
-                    }
-                    else
-                    {
-                        MessageBox.Show(this, $@"{_deleteWorktreeFailedText.Text}: {workTree.Path}{Environment.NewLine}{errorMessage}", TranslatedStrings.Error,
-                            MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    }
-                }
+        private bool CanDeleteSelectedWorkspace()
+            => CanActOnSelectedWorkspace(out _) && Worktrees.SelectedRows[0].Index != 0;
+
+        private bool CanActOnSelectedWorkspace(out WorkTree workTree)
+        {
+            workTree = null;
+
+            if (_worktrees == null || _worktrees.Count == 1 || Worktrees.SelectedRows.Count == 0)
+            {
+                return false;
             }
+
+            workTree = _worktrees[Worktrees.SelectedRows[0].Index];
+
+            if (workTree.IsDeleted)
+            {
+                return false;
+            }
+
+            return !IsCurrentlyOpenedWorktree(workTree);
+        }
+
+        private bool IsCurrentlyOpenedWorktree(WorkTree workTree)
+            => new DirectoryInfo(UICommands.GitModule.WorkingDir).FullName.TrimEnd('\\') == new DirectoryInfo(workTree.Path).FullName.TrimEnd('\\');
+
+        private void buttonCreateNewWorktree_Click(object sender, EventArgs e)
+        {
+            SwitchToCreateWorktreeForm = true;
+            DialogResult = DialogResult.Cancel;
```

Comment:
I would rather keep this dialog open and open the FormCreateWorktree directly.
The reaction to `Open the new worktree after creation` can reuse `buttonOpenSelectedWorktree`.